src/sync.rs

```diff
@@ -281,6 +286,137 @@ impl<T: StableDeref> FrozenVec<T> {
     // TODO add more
 }
 
+/// Append-only threadsafe version of `std::vec::Vec` where
+/// insertion does not require mutable access.
+/// Does not have locks, only allows `Copy` types and will
+/// spinlock on contention. The spinlocks are really rare as
+/// they only happen on reallocation due to a push going over
+/// the capacity.
+pub struct LockFreeFrozenVec<T: Copy> {
+    data: AtomicPtr<T>,
+    len: AtomicUsize,
+    cap: AtomicUsize,
+}
+
+impl<T: Copy> Drop for LockFreeFrozenVec<T> {
+    fn drop(&mut self) {
+        let ptr = self.aquire();
+        let cap = self.cap.load(Ordering::Acquire);
+        let num_bytes = std::mem::size_of::<T>() * cap;
+        let align = std::mem::align_of::<T>();
+        let layout = Layout::from_size_align(num_bytes, align).unwrap();
+        unsafe {
+            std::alloc::dealloc(ptr.cast(), layout);
+        }
+    }
+}
+
+impl<T: Copy> Default for LockFreeFrozenVec<T> {
+    fn default() -> Self {
+        Self {
+            data: AtomicPtr::new(Box::into_raw(Box::new([Self::UNINIT; 128])).cast()),
+            len: AtomicUsize::new(0),
+            cap: AtomicUsize::new(128),
+        }
+    }
+}
+
+impl<T: Copy> LockFreeFrozenVec<T> {
+    const UNINIT: MaybeUninit<T> = MaybeUninit::uninit();
+
+    pub fn new() -> Self {
```

Comment:
nit: please document this as doing an allocation, and perhaps add an `::empty()` constructor as well

Reply:
I added a `with_capacity` and removed the allocation in new/default